### PR TITLE
docs: Add Rules page intro and content tweaks

### PR DESCRIPTION
**Pull request by bpmutter**
#### Prerequisites checklist

- [x] I have read the [contributing guidelines](https://github.com/eslint/eslint/blob/HEAD/CONTRIBUTING.md).

#### What is the purpose of this pull request? (put an "X" next to an item)

[x] Documentation update
[ ] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/HEAD/templates/rule-change-proposal.md))
[ ] Add autofix to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

#### What changes did you make? (Give an overview)

Add short introduction to the Configuring Rules page and minor content tweaks throughout the page.

Part of https://github.com/eslint/eslint/issues/16473

### CODE REVIEW

File: docs/src/user-guide/configuring/rules.md [hunk 1]
```diff
@@ -278,7 +290,7 @@ To disable all inline config comments, use the `noInlineConfig` setting. For exa
 }
 ```
 
-This setting is similar to [--no-inline-config](../command-line-interface#--no-inline-config) CLI option.
+You can also use the [--no-inline-config](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configuration.
```
Comment (amareshsm) [inline docs/src/user-guide/configuring/rules.md:293]:
```suggestion
You can also use the [--no-inline-config](../command-line-interface#--no-inline-config) CLI option to disable rule comments, in addition to other in-line configurations.
```

Comment (nzakas) [reply docs/src/user-guide/configuring/rules.md:293]: I believe the singular is correct here but will let @bpmutter comment.

File: docs/src/user-guide/configuring/rules.md [hunk 2]
```diff
@@ -117,7 +127,9 @@ rules:
   plugin1/rule1: error
 ```
 
-In these configuration files, the rule `plugin1/rule1` comes from the plugin named `plugin1`. You can also use this format with configuration comments, such as:
+In these configuration files, the rule `plugin1/rule1` comes from the plugin named `plugin1`.
```
Comment (nzakas) [inline docs/src/user-guide/configuring/rules.md:130]: For clarity:

```suggestion
In these configuration files, the rule `plugin1/rule1` comes from the plugin named `plugin1`, which is contained in an npm package named `eslint-plugin-plugin1`.
```

File: docs/src/user-guide/configuring/rules.md [hunk 3]
```diff
@@ -9,9 +9,13 @@ eleventyNavigation:
 
 ---
 
-## Configuring Rules
+Rules are the core building block of ESLint. A rule validates if your code meets a certain expectation, and what to do if it does not meet that expectation. Rules can also contain additional configuration options specific to that rule.
 
-ESLint comes with a large number of built-in rules and you can add more rules through plugins. You can modify which rules your project uses either using configuration comments or configuration files. To change a rule setting, you must set the rule ID equal to one of these values:
+ESLint comes with a large number of [built-in rules](../../rules/) and you can add more rules through plugins. You can modify which rules your project uses with either configuration comments or configuration files.
+
+## Rule Settings
```
Comment (nzakas) [inline docs/src/user-guide/configuring/rules.md:16]: Settings are a specific thing in ESLint; this is a severity.

```suggestion
## Rule Severities
```

File: docs/src/user-guide/configuring/rules.md [hunk 4]
```diff
@@ -9,9 +9,13 @@ eleventyNavigation:
...
+
+To change a rule setting, you must set the rule ID equal to one of these values:
```
Comment (nzakas) [inline docs/src/user-guide/configuring/rules.md:18]:
```suggestion
To change a rule's severity, set the rule ID equal to one of these values:
```

File: docs/src/user-guide/configuring/rules.md [hunk 5]
```diff
@@ -86,7 +92,11 @@ rules:
     - double
 ```
 
-To configure a rule which is defined within a plugin you have to prefix the rule ID with the plugin name and a `/`. For example:
+## Rules from Plugins
+
+To configure a rule which is defined within a plugin you have to prefix the rule ID with the plugin name and `/`.
```
Comment (nzakas) [inline docs/src/user-guide/configuring/rules.md:97]: Removing one "w" word. :)

```suggestion
To configure a rule that is defined within a plugin, prefix the rule ID with the plugin name and `/`.
```

**Review by nzakas — CHANGES_REQUESTED** [review]
Thanks for cleaning this up, it's much better.

I left a few suggestions. I'd like to remove "you must" and "you have to" throughout, as I think they're a little too aggressive on their own.

File: docs/src/user-guide/configuring/rules.md (hunk 1 above)
Comment (snitin315) [reply docs/src/user-guide/configuring/rules.md:293]: Hmm, this one seems tricky. Grammarly shows plural

Screenshot 2022-11-15 at 7 17 35 AM

Comment (harish-sethuraman) [reply docs/src/user-guide/configuring/rules.md:293]: I also feel `configuration` to be correct here? as we have only one configuration?

Comment (amareshsm) [reply docs/src/user-guide/configuring/rules.md:293]: `In addition to other` which makes it plural.

In the CLI option, we can pass many configurations, right? eg:

```
eslint index.js --no-eslintrc --env "es6"  --parser-options "{ecmaVersion: 2018}" --no-inline-config
```

Comment (harish-sethuraman) [reply docs/src/user-guide/configuring/rules.md:293]: In this place we were meaning `inline-config`. I guess we specify multiple inline-configs as one inline configuration? (`/* eslint indent: "error", curly: "error" */`)

Comment (bpmutter) [reply docs/src/user-guide/configuring/rules.md:293]: generally the word 'configuration' is only singular - https://www.wordhippo.com/what-is/the-plural-of/configuration.html

'configurations' usually only when there are multiple different *sets* of configurations. 

in this case, i think both could be correct. but singular reads better to me.

**Comment by bpmutter** [comment]
applied suggestions from code review!

**Review by snitin315 — APPROVED** [review]
LGTM, thank you!

**Review by nzakas — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**